enhancements/network/ovn-k-localnet-api.md

```diff
@@ -0,0 +1,514 @@
+
+| Field name          | Description                                                                                                                                                                                                                                                                                           | optional |
+|---------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|
+| Role                | Select the network role in the pod.                                                                                                                                                                                                                                                                   | No       |
+| PhysicalNetworkName | The OVN bridge mapping network name is configured on the node.                                                                                                                                                                                                                                        | No       |
+| MTU                 | The maximum transmission unit (MTU).                                                                                                                                                                                                                                                                  | Yes      |
+| VLAN                | The network VLAN ID.                                                                                                                                                                                                                                                                                  | Yes      |
+| Subnets             | List of CIDRs used for the pod network across the cluster.Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed. The format should match standard CIDR notation (for example, "10.128.0.0/16"). This field must be omitted if `ipam.mode` is `Disabled`. | Yes      |
+| ExcludeSubnets      | List of CIDRs removed from the specified CIDRs in `subnets`.The format should match standard CIDR notation (for example, "10.128.0.0/16"). This field must be omitted if `subnetes` is unset or `ipam.mode` is `Disabled`.                                                                            | Yes      |
+| IPAM                | Contains IPAM-related configuration for the network. When ipam.lifecycle=Persistent enable workloads have persistent IP addresses. For example: Virtual Machines will have the same IP addresses along their lifecycle (stop, start migration, reboots).                                              | Yes      |
```

Comment:
I'm conflicted. Should we actually move the subnets + excludesubnets attributes to IPAM ? ... 

It kind of makes sense ... but would make it diverge from L2.

Reply:
we definitely want all network types to be consistent, and we discussed this (more than once :D) before.
I think one of the reason to leave it separate was our plan to use subnets field for egress setup but with disabled ipam, so subnets isn't necessarily a part of ipam